scikeras/utils/transformers.py

```diff
@@ -0,0 +1,201 @@
+                )
+            X = X.reshape(-1, 1)
+        return X
+
+    def inverse_transform(self, X: np.ndarray) -> np.ndarray:
+        if self.should_transform_:
+            if X.ndim != 2:
+                raise ValueError(
+                    f"Expected a 2D array for `X`, but got a"
+                    f" {X.ndim}D array instead."
+                )
+            if X.shape[1] != 1:
+                raise ValueError(
+                    f"Expected `X.shape[1] == 1`"
+                    f" but got `X.shape[1] == {X.shape}` instead."
+                )
+            X = np.squeeze(X, axis=1)
+        return X
+
+
+class ClassifierLabelEncoder(BaseEstimator, TransformerMixin):
+    """Default target transformer for KerasClassifier.
+    """
+
+    def __init__(self, loss=None):
+        self.loss = loss
+
+    def fit(self, X: np.ndarray) -> "ClassifierLabelEncoder":
+        y = X  # rename for clarity, the input is always expected to be a target `y`
+        target_type = type_of_target(y)
+        encoders = {
+            "binary": make_pipeline(
+                Ensure2DTransformer(), OrdinalEncoder(dtype=np.float32),
+            ),
+            "multiclass": make_pipeline(
+                Ensure2DTransformer(), OrdinalEncoder(dtype=np.float32),
+            ),
+            "multiclass-multioutput": FunctionTransformer(),
+            "multilabel-indicator": FunctionTransformer(),
+        }
+        if is_categorical_crossentropy(self.loss):
+            encoders["multiclass"] = make_pipeline(
+                Ensure2DTransformer(), OneHotEncoder(sparse=False, dtype=np.float32),
+            )
+        if target_type not in encoders:
+            raise ValueError(
+                f"Unknown label type: {target_type}."
+                "\n\nTo implement support, subclass KerasClassifier and override"
+                " `target_transformer` with a transformer that supports this"
+                " label type."
+            )
+        self.final_encoder_ = encoders[target_type].fit(y)
+
+        if target_type == "multilabel-indicator" and np.all(np.sum(y, axis=1) == 1):
+            target_type = "multiclass-onehot"
+            self.n_outputs_ = None
+            self.model_n_outputs_ = None
+            self.y_dtype_ = y.dtype
+        else:
+            self.n_outputs_ = 1
+            self.model_n_outputs_ = 1
+            self.y_dtype_ = y.dtype
+
+        self._target_type = target_type
+
+        if target_type in ["binary", "multiclass"]:
+            self.classes_ = self.final_encoder_[1].categories_[0]
+            self.n_classes_ = self.classes_.size
+        elif target_type == "multiclass-onehot":  # one-hot encoded multiclass
+            self.classes_ = np.arange(0, y.shape[1])
+            self.n_classes_ = y.shape[1]
+        else:
+            self.classes_ = None
+            self.n_classes_ = None
+        self.n_outputs_ = 1
+        self.model_n_outputs_ = 1
```

Comment:
I would only set class attributes once; `self.n_outputs_` and `self.model_n_outputs_` are also set on line 86 (14 lines above).

Reply:
I'll clean this up.

What do you think we should do with these attributes for the pure `multilabel-indicator` and `multiclass-multioutput` target types? Set them to `None`?

Reply:
Cleaned up the duplication. Closing discussion on attributes in favor of https://github.com/adriangb/scikeras/pull/88#discussion_r502987233